tests/test_jira.py

```diff
@@ -0,0 +1,422 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+#
+# Authors:
+#     Alberto Martín <alberto.martin@bitergia.com>
+#     Quan Zhou <quan@bitergia.com>
+#
+
+import argparse
+import json
+import unittest
+import shutil
+import sys
+import tempfile
+
+import httpretty
+
+if not '..' in sys.path:
+    sys.path.insert(0, '..')
+
+from perceval.cache import Cache
+from perceval.errors import BackendError, CacheError, ParseError
+from perceval.backends.jira import Jira, JiraClient, JiraCommand
+from perceval.utils import str_to_datetime
+
+
+JIRA_SERVER_URL = 'http://example.com'
+JIRA_SEARCH_URL = JIRA_SERVER_URL + '/rest/api/2/search'
+
+
+def read_file(filename, mode='r'):
+    with open(filename, mode) as f:
+        content = f.read()
+    return content
+
+
+class TestJiraBAckend(unittest.TestCase):
+    """Jira backend tests"""
+
+    @httpretty.activate
+    def test_fetch(self):
+        """Test whether a list of issues is returned"""
+
+        requests = []
+
+        bodies_json = [read_file('data/jira/jira_issues_page_1.json'),
+                       read_file('data/jira/jira_issues_page_2.json')]
+
+        def request_callback(method, uri, headers):
+            body = bodies_json.pop(0)
+            requests.append(httpretty.last_request())
+            return (200, headers, body)
+
+        httpretty.register_uri(httpretty.GET,
+                               JIRA_SEARCH_URL,
+                               responses=[httpretty.Response(body=request_callback) \
+                                          for _ in range(2)])
+
+        jira = Jira(JIRA_SERVER_URL)
+
+        issues = [issue for issue in jira.fetch()]
+
+        expected_req = [{
+                            'expand': ['renderedFields,transitions,operations,changelog'],
+                            'jql': [' updated > 0'],
+                            'startAt': ['0']
+                        },
+                        {
+                            'expand': ['renderedFields,transitions,operations,changelog'],
+                            'jql': [' updated > 0'],
+                            'startAt': ['2']
+                        }]
+
+        self.assertEqual(requests[0].method, 'GET')
+        self.assertRegex(requests[0].path, '/rest/api/2/search')
+        self.assertDictEqual(requests[0].querystring, expected_req[0])
+
+        self.assertEqual(requests[1].method, 'GET')
+        self.assertRegex(requests[1].path, '/rest/api/2/search')
+        self.assertDictEqual(requests[1].querystring, expected_req[1])
+
+        self.assertEqual(len(issues), 3)
+
+        self.assertEqual(issues[0]['perceval_version'], '0.1.0')
```

Comment:
None of these (`perceval_version`, `perceval_backend` and `backend_name`) need to be tested. These parameters are tested on `test_backend.py`. Instead, check one of the contents of the issue, for instance the id of the bug.